collect_app/src/main/java/org/odk/collect/android/logic/AuditEvent.java

```diff
@@ -273,4 +272,35 @@ private String getEscapedValueForCsv(String value) {
 
         return "\"" + value + "\"";
     }
+
+    /**
+     * Get the XPath path of the node at a particular {@link FormIndex}.
+     *
+     * Differs from {@link TreeReference#toString()} in that position predicates are only
+     * included for repeats. For example, given a group named {@code my-group} that contains a
+     * repeat named {@code my-repeat} which in turn contains a question named {@code my-question},
+     * {@link TreeReference#toString()} would return paths that look like
+     * {@code /my-group[1]/my-repeat[3]/my-question[1]}. In contrast, this method would return
+     * {@code /my-group/my-repeat[3]/my-question}.
+     *
+     * TODO: consider moving to {@link FormIndex}
+     */
+    private static String getXPathPath(FormIndex formIndex) {
```

Comment:
It makes sense me that this should be either a `private` (`static`) method in `AuditEvent` as it doesn't seem to be used outside to the object and it hasn't been tested in isolation. I'm guessing having it `static` is because you plan to move it to `FormIndex`?

Reply:
Yes, exactly.